lib/hammer_cli_foreman/commands.rb

```diff
@@ -311,6 +311,86 @@ def should_retrieve_all?
 
   end
 
+  class ListSearchCommand < ListCommand
+    def self.search_resource(res, action = :index)
+      resource res, action
+      default_search_options
+    end
+
+    def self.default_search_options
+      option("--#{module_resource.singular_name}-id", "ID", _("%s Id") % module_resource.singular_name)
```

Comment:
Some module resources could look like ```:ansible_roles```, so this will lead to ```--ansible_role-id``` which is incorrect.

How about using .tr('_', '-') like ```"--#{module_resource.singular_name}-id".tr('_', '-')```? So there is no need to redefine this method in plugins.